runtime/lua/vim/diagnostic.lua

```diff
@@ -2591,4 +2591,18 @@ function M.fromqflist(list)
   return diagnostics
 end
 
+function M.status()
+  local counts = M.count(0)
+  local user_signs = vim.tbl_get(M.config(), 'signs', 'text') or {}
```

Comment:
```suggestion
  local user_signs = vim.tbl_get(M.config() --[[@as vim.diagnostic.Opts]], 'signs', 'text') or {}
```
To satisfy CI. But M.config() should probably have an overload to handle both signatures `fun(vim.diagnostic.Opts): nil` and `fun(): vim.diagnostic.Opts`.